### PR TITLE
Goodbye "goto relex" (refactor)

### PR DESCRIPTION
**Pull request by MrAnno**
This PR contains equivalent transformations only (I hope so).

Extracted methods:
- `cfg_lexer_lex_next_token()`
- `cfg_lexer_append_preprocessed_output()`
- `cfg_lexer_parse_and_run_block_generator()`
- `cfg_lexer_parse_include()`
- `cfg_lexer_preprocess()`

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/cfg-lexer.c [hunk 1]
```diff
@@ -1069,6 +1047,52 @@ relex:;
...
+      if (cfg_lexer_get_context_type(self) == LL_CONTEXT_BLOCK_CONTENT)
+        cfg_lexer_start_block_state(self, "{}");
+      else if (cfg_lexer_get_context_type(self) == LL_CONTEXT_BLOCK_ARG)
+        cfg_lexer_start_block_state(self, "()");
+
+      tok = cfg_lexer_lex_next_token(self, yylval, yylloc);
+      cfg_lexer_append_preprocessed_output(self, self->token_pretext->str);
+    }
+
+  if (!cfg_lexer_process_token(self, tok, yylval, yylloc, &relex_required))
```
Comment (Kokan) [inline lib/cfg-lexer.c]: This function is a little confusing for me. It has a return value, and other bool value via parameter (`relex_required`), and those two variable describes what the next action should be. Those two variable are tangled, but the code does not represents it.
Please if possible couple those variables.

File: lib/cfg-lexer.c [hunk 2]
```diff
@@ -1050,13 +1047,49 @@ cfg_lexer_lex(CfgLexer *self, YYSTYPE *yylval, YYLTYPE *yylloc)
...
+          if (cfg_lexer_get_context_type(self) == LL_CONTEXT_BLOCK_CONTENT)
+            cfg_lexer_start_block_state(self, "{}");
+          else if (cfg_lexer_get_context_type(self) == LL_CONTEXT_BLOCK_ARG)
+            cfg_lexer_start_block_state(self, "()");
+
+          tok = cfg_lexer_lex_next_token(self, yylval, yylloc);
+          cfg_lexer_append_preprocessed_output(self, self->token_pretext->str);
         }
+
+      if (!cfg_lexer_process_token(self, tok, yylval, yylloc, &lex_again))
```
Comment (Kokan) [inline lib/cfg-lexer.c]: This function is a little confusing for me. It has a return value, and other bool value via parameter (`relex_required`), and those two variable describes what the next action should be. Those two variable are tangled, but the code does not represents it.
Please if possible couple those variables.

Comment (MrAnno) [reply lib/cfg-lexer.c]: OK, I have an idea.

Comment (MrAnno) [reply lib/cfg-lexer.c]: Fixed, thanks.

**Comment by MrAnno** [comment]
@szemere @Kokan: I've pushed 2 additional commits (and left the rest intact).

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/cfg-lexer.c [hunk 3]
```diff
@@ -1082,10 +1085,11 @@ cfg_lexer_lex(CfgLexer *self, YYSTYPE *yylval, YYLTYPE *yylloc)
           cfg_lexer_append_preprocessed_output(self, self->token_pretext->str);
         }
 
-      if (!cfg_lexer_preprocess(self, tok, yylval, yylloc, &lex_again))
+      preprocess_result = cfg_lexer_preprocess(self, tok, yylval, yylloc);
+      if (preprocess_result == CLPR_ERROR)
```
Comment (szemere) [inline lib/cfg-lexer.c]: note: I think we can avoid a condition inside the loop, and move it afterwards.

**Comment by lbudai** [comment]
@MrAnno : thanks!

**Comment by bazsi** [comment]
\o/

**Pull request merged**